src/main/kotlin/io/snyk/plugin/services/SnykAnalyticsService.kt

```diff
@@ -42,29 +85,119 @@
         }
         val userId = service<SnykApiService>().userId
         if (userId == null) {
-            log.warn("Not able to obtain User public id.")
+            log.warn("Not able to obtain user public id.")
             return ""
         }
         return userId
     }
 
     fun identify() {
-        catchAll(log, "identify") { segment.identify() }
+        if (!canReportEvents() || userId.isBlank()) {
```

Comment:
minor: to shorter code, those ifs could be reverted or even moved into catchAll fun